coffee-chats/src/main/webapp/src/entity/Requests.ts

```diff
@@ -0,0 +1,157 @@
+export interface ChatRequest {
+  tags: string[],
+  dateRanges: DateRange[],
+  minPeople: number,
+  maxPeople: number,
+  durationMins: number,
+  matchRandom: boolean,
+  matchRecents: boolean,
+  userId: string,
+  requestId: number
+};
+
+export type CompletedRequest = ExpiredRequest | MatchedRequest;
+
+export interface MatchedRequest {
+  matched: boolean,
```

Comment:
A lot of repeated code, maybe create an interface with common fields and inherit from it?

Reply:
I opted to not do this initially for clarity and separation, but the majority of two of the request types are duplicated. I have refactored this now.